homeassistant/components/mqtt/__init__.py

```diff
@@ -226,6 +226,16 @@
     }
 )
 
+CONFIG_ENTRY_CONFIG_KEYS = [
+    CONF_BIRTH_MESSAGE,
+    CONF_BROKER,
+    CONF_DISCOVERY,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    CONF_WILL_MESSAGE,
+]
+
```

Comment:
If we add a default value (dict) we can warn that a setting was removed that has not a default value?